lib/srv/db/opensearch_test.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package db
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/defaults"
+	libevents "github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	"github.com/gravitational/teleport/lib/srv/db/opensearch"
+)
+
+func registerTestOpenSearchEngine() {
+	common.RegisterEngine(newTestOpenSearchEngine, defaults.ProtocolOpenSearch)
+}
+
+func newTestOpenSearchEngine(ec common.EngineConfig) common.Engine {
+	staticAWSCredentials := func(client.ConfigProvider, time.Time, string, string, string) *credentials.Credentials {
+		return credentials.NewStaticCredentials("AKIDl", "SECRET", "SESSION")
+	}
+
+	return &opensearch.Engine{
+		EngineConfig: ec,
+		// inject mock AWS credentials.
+		GetSigningCredsFn: staticAWSCredentials,
+	}
+}
+
+func TestAccessOpenSearch(t *testing.T) {
+	ctx := context.Background()
+	testCtx := setupTestContext(ctx, t, withOpenSearch("OpenSearch"))
+	go testCtx.startHandlingConnections()
+
+	tests := []struct {
+		desc         string
+		user         string
+		role         string
+		allowDbUsers []string
+		dbUser       string
+		err          bool
+	}{
+		{
+			desc:         "has access to all database names and users",
+			user:         "alice",
+			role:         "admin",
+			allowDbUsers: []string{types.Wildcard},
+			dbUser:       "opensearch",
+		},
+		{
+			desc:         "has access to nothing",
+			user:         "alice",
+			role:         "admin",
+			allowDbUsers: []string{},
+			dbUser:       "opensearch",
+			err:          true,
+		},
+		{
+			desc:         "no access to users",
+			user:         "alice",
+			role:         "admin",
+			allowDbUsers: []string{},
+			dbUser:       "opensearch",
+			err:          true,
+		},
+		{
+			desc:         "access allowed to specific user/database",
```

Comment:
Correct me if I'm wrong but In OpenSearch integration we don't have restrition check for database name so the test name should be: 
```suggestion
			desc:         "access allowed to specific user",
```

Reply:
Done.